pkg/yurtmanager/webhook/yurtappoverrider/v1alpha1/yurtappoverrider_validation.go

```diff
@@ -0,0 +1,109 @@
+)
+
+// ValidateCreate implements webhook.CustomValidator so a webhook will be registered for the type.
+func (webhook *YurtAppOverriderHandler) ValidateCreate(ctx context.Context, obj runtime.Object) error {
+	overrider, ok := obj.(*v1alpha1.YurtAppOverrider)
+	if !ok {
+		return apierrors.NewBadRequest(fmt.Sprintf("expected a YurtAppOverrider but got a %T", obj))
+	}
+
+	// validate
+	if err := webhook.validateOneToOne(ctx, overrider); err != nil {
+		return err
+	}
+	if err := webhook.validateStar(overrider); err != nil {
+		return err
+	}
+	return nil
+}
+
+// ValidateUpdate implements webhook.CustomValidator so a webhook will be registered for the type.
+func (webhook *YurtAppOverriderHandler) ValidateUpdate(ctx context.Context, oldObj, newObj runtime.Object) error {
+	_, ok := newObj.(*v1alpha1.YurtAppOverrider)
+	if !ok {
+		return apierrors.NewBadRequest(fmt.Sprintf("expected a YurtAppOverrider but got a %T", newObj))
+	}
+	newOverrider, ok := oldObj.(*v1alpha1.YurtAppOverrider)
+	if !ok {
+		return apierrors.NewBadRequest(fmt.Sprintf("expected a YurtAppOverrider} but got a %T", oldObj))
+	}
+
+	// validate
+	if err := webhook.validateOneToOne(ctx, newOverrider); err != nil {
+		return err
+	}
+	if err := webhook.validateStar(newOverrider); err != nil {
+		return err
+	}
+	return nil
+}
+
+// ValidateDelete implements webhook.CustomValidator so a webhook will be registered for the type.
+func (webhook *YurtAppOverriderHandler) ValidateDelete(_ context.Context, obj runtime.Object) error {
+	_, ok := obj.(*v1alpha1.YurtAppOverrider)
+	if !ok {
+		return apierrors.NewBadRequest(fmt.Sprintf("expected a YurtAppOverrider but got a %T", obj))
+	}
+	// validate
+	return nil
+}
+
+// YurtAppOverrider and YurtAppSet are one-to-one relationship
+func (webhook *YurtAppOverriderHandler) validateOneToOne(ctx context.Context, yurtAppOverrider *v1alpha1.YurtAppOverrider) error {
+	app := yurtAppOverrider.Subject
+	var allOverriderList v1alpha1.YurtAppOverriderList
+	if err := webhook.Client.List(ctx, &allOverriderList, client.InNamespace(yurtAppOverrider.Namespace)); err != nil {
+		klog.Info("error in listing YurtAppOverrider")
+		return err
+	}
+	var overriderList = v1alpha1.YurtAppOverriderList{}
+	for _, overrider := range overriderList.Items {
+		if overrider.Subject.Kind == app.Kind && overrider.Name == app.Name && overrider.APIVersion == app.APIVersion {
+			overriderList.Items = append(overriderList.Items, overrider)
+		}
+	}
+	if len(overriderList.Items) > 0 {
+		return fmt.Errorf("only one YurtAppOverrider can be bound into one YurtAppSet")
+	}
+	return nil
+}
+
+// Verify that * and other pools are not set at the same time
+func (webhook *YurtAppOverriderHandler) validateStar(yurtAppOverrider *v1alpha1.YurtAppOverrider) error {
+	for _, entry := range yurtAppOverrider.Entries {
+		for _, pool := range entry.Pools {
+			if pool == "*" && len(entry.Pools) > 1 {
+				return fmt.Errorf("pool can't be '*' when other pools are set")
```

Comment:
would you like to explain the reason of this `if condition` --> if pool == "*" && len(entry.Pools) > 1